trl/trainer/ppo_trainer.py

```diff
@@ -444,7 +445,11 @@ def _generate_batched(
             generations = self.accelerator.unwrap_model(self.model).generate(**padded_inputs, **generation_kwargs)
 
             for generation, mask in zip(generations, padded_inputs["attention_mask"]):
-                output = generation[(1 - mask).sum() :]  # remove padding
+                if not self.is_encoder_decoder:
+                    output = generation[(1 - mask).sum() :]  # remove padding
+                else:
+                    output = generation
```

Comment:
maybe we could remove the special token here?

Reply:
I am not sure if this is correct as the token is removed here right after: https://github.com/lvwerra/trl/blob/ed87942a47f26d15e823ca7674737be02e48cc0a/trl/trainer/ppo_trainer.py#L832 
Also made a run with `generation[1:]` and the KL becomes negative: https://wandb.ai/younesbelkada/trl/runs/vjbydeqv -  so I think we shouldn't remove the special token here

Reply:
Is that not suspicious that such a slight change breaks the training? If no then why is that expected? I'm asking this as I myself am having trouble with my t5 model and negative kl divergence even on v.0.4.1 release.